### PR TITLE
Deal with warnings attached to value when making widgets.

### LINKED ISSUE #5893: Dynamic widget visualization throws an error
On latest engine versions, the visualization that is responsible for widget data throws an error. In effect, the dynamic widgets never how up. The error is reported in the console as a visualization execution error.

<details>
<summary>Reported visualization error: <code>Method `map` of Function could not be found.</code></summary>

```json
{
  "jsonrpc": "2.0",
  "method": "executionContext/visualisationEvaluationFailed",
  "params": {
    "contextId": "6d8fccce-38b3-495a-87ed-4af06a67ddb5",
    "visualisationId": "41fec3bf-eebc-44d7-a185-bca5753a33f0",
    "expressionId": "c85c515e-cc9e-4022-a826-f4faeb6db66d",
    "message": "Method `map` of Function could not be found.",
    "diagnostic": {
      "kind": "Error",
      "message": "Method `map` of Function could not be found.",
      "path": {
        "rootId": "19e3e980-e2f7-4b4c-bf20-5753a2e4aaf7",
        "segments": [
          "src",
          "Main.enso"
        ]
      },
      "location": {
        "start": {
          "line": 17,
          "character": 16
        },
        "end": {
          "line": 17,
          "character": 56
        }
      },
      "expressionId": "8148e959-14dc-4530-b0d8-58e6b832b0e2",
      "stack": [
        {
          "functionName": "Main.main",
          "path": {
            "rootId": "19e3e980-e2f7-4b4c-bf20-5753a2e4aaf7",
            "segments": [
              "src",
              "Main.enso"
            ]
          },
          "location": {
            "start": {
              "line": 17,
              "character": 16
            },
            "end": {
              "line": 17,
              "character": 56
            }
          },
          "expressionId": "8148e959-14dc-4530-b0d8-58e6b832b0e2"
        }
      ]
    }
  }
}
```
</details>

Affected engine release: `2023.1.1-nightly.2023.3.9`
Known good engine release: `2023.1.1-nightly.2023.3.4`

|Missing widget on `2023.1.1-nightly.2023.3.9`|Widget on `2023.1.1-nightly.2023.3.4`|
|-|-|
|No drop down for Table.at|Dropdown with column names for Table.at|

Issue orginally reported in https://github.com/enso-org/enso/pull/5852#issuecomment-1462199693

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Fixes InvokeCallableNode to support warnings.
- Strips warnings from annotations in `get_widget_json`.
- Remove `get_full_annotations_json`.
- Fix warnings on Dialect.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java [hunk 1]
```diff
@@ -251,6 +253,33 @@ public Object invokeDynamicSymbol(
...
 
+  @Specialization(guards = "warnings.hasWarnings(warning)")
+  public Object invokeWarnings(
+      Object warning,
+      VirtualFrame callerFrame,
+      State state,
+      Object[] arguments,
+      @CachedLibrary(limit = "3") WarningsLibrary warnings) {
+    try {
+      var result = execute(
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java]: It'd be better to delegate to another node instance. Add `@Cached InvokeCallableNode delegate` to argument list.

**Review by JaroslavTulach — APPROVED** [review]
Is there a test that fails without the changes in `InvokeCallableNode`? It'd be desirable to have it even in `engine/runtime` project.

File: engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java (hunk 1 above)
Comment (jdunkerley) [reply engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java]: added but get a compile error:

```
[error] C:\Repos\Enso\ide\engine\runtime\src\main\java\org\enso\interpreter\node\callable\InvokeCallableNode.java:266:1:  error: Error parsing expression 'create()': The method create(CallArgumentInfo[], DefaultsExecutionMode, ArgumentsExecutionMode) in the type InvokeCallableNodeGen is not applicable for the arguments .
[error]       @Cached InvokeCallableNode invokeCallableNode,
[error]       ^warning: Unclosed files for the types 'org.enso.interpreter.runtime.type.ConstantsGen'; these types will not undergo annotation processing
```

**Comment by jdunkerley** [comment]
Still one issue to solve: 
image

The IDE isn't adding the parameters to functions with a warning attached.

File: engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java [hunk 2]
```diff
@@ -251,6 +257,34 @@ public Object invokeDynamicSymbol(
...
+                  arguments);
+
+      Warning[] extracted = warnings.getWarnings(warning, null);
+      if (result instanceof DataflowError) {
+        return result;
+      } else if (result instanceof WithWarnings withWarnings) {
+        return withWarnings.prepend(extracted);
+      } else {
+        return new WithWarnings(result, extracted);
+      }
```
Comment (radeusgd) [inline engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java:302]: This feels like an universal piece of logic when we are attaching warnings to a result which may already also contain new warnings.

Is similar logic used elsewhere?
Maybe we can refactor it into a separate function `WithWarnings::attachToObject`?

**Review by radeusgd — APPROVED** [review]

**Comment by jdunkerley** [comment]
> Is there a test that fails without the changes in `InvokeCallableNode`? It'd be desirable to have it even in `engine/runtime` project.

Happy to add a test - any where specific to put it?
I couldn't obviously see a place for warning test etc

File: engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java (hunk 2 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java:302]: Having a `Node` to handle extraction and later addition of warnings would be great. We had this problem so many times already.

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**